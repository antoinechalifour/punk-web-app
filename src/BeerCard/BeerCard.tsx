import React from "react";

import placeholder from "./placeholder.png";
import { Card, CardContent } from "../ui/Card";
import {
  Wrapper,
  InnerLayout,
  Image,
  Details,
  Name,
  TagLine,
  Link
} from "./styles";

export interface BeerCardProps {
  imageUrl: string;
  name: string;
  tagline: string;
  id: number;
}

export const BeerCard: React.FunctionComponent<BeerCardProps> = ({
  id,
  imageUrl,
  name,
  tagline
}) => (
  <Wrapper>
    <Card>
      <CardContent>
        <InnerLayout>
          <Image
            src={imageUrl}
            alt={`Bottle of ${name}`}
            placeholderSrc={placeholder}
          />
          <Details>
            <Name>{name}</Name>
            <TagLine>{tagline}</TagLine>
            <Link to={`/beers/${id}`}>Read more</Link>
          </Details>
        </InnerLayout>
      </CardContent>
    </Card>
  </Wrapper>
);
