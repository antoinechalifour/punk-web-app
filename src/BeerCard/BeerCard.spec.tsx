import { shallow } from "enzyme";
import React from "react";

import { BeerCard, BeerCardProps } from "./BeerCard";

it("should match snapshot", () => {
  const props: BeerCardProps = {
    imageUrl: "http://example.com/picture.png",
    id: 1,
    name: "Punk IPA",
    tagline: "Post modern classic"
  };

  const wrapper = shallow(<BeerCard {...props} />);

  expect(wrapper).toMatchSnapshot();
});
