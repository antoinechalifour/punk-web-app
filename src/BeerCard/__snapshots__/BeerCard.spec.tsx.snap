// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.img
          alt="Bottle of Punk IPA"
          src="http://example.com/picture.png"
        />
        <styled.div>
          <styled.p>
            Punk IPA
          </styled.p>
          <styled.p>
            Post modern classic
          </styled.p>
          <Styled(Link)
            to="/beers/1"
          >
            Read more
          </Styled(Link)>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
